### PR TITLE
chore(website): set up env based search indexing

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Change Algolia indexing so it only happens in the production deployment environment

### CODE REVIEW

File: packages/paste-website/gatsby-config.js [hunk 1]
```diff
@@ -187,15 +189,24 @@ module.exports = {
...
+  options: {
+    appId: process.env.GATSBY_ALGOLIA_APP_ID,
+    apiKey: process.env.GATSBY_ALGOLIA_ADMIN_KEY,
+    indexName: process.env.GATSBY_ALGOLIA_INDEX_NAME,
+    queries,
+    chunkSize: 10000,
+  },
+};
+
+const activeEnv = process.env.DEPLOYMENT_ENV;
```
Comment (TheSisb) [inline packages/paste-website/gatsby-config.js:206]: Where is DEPLOYMENT_ENV being set?

Comment (richbachman) [reply packages/paste-website/gatsby-config.js:206]: In our Vercel environment variables

File: packages/paste-website/gatsby-config.js [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
 
-// const queries = require('./src/utils/algolia');
+const queries = require('./src/utils/algolia');
 
 require('dotenv').config();
 
-module.exports = {
+const gatsbyConfig = {};
+
+gatsbyConfig.data = {
```
Comment (TheSisb) [inline packages/paste-website/gatsby-config.js:13]: Not sure why we need this `data` key. Can we do
```
const gatsbyConfig = {
  siteMetadata: ...
}

if (proces.env.DEPLOYMENT_ENV === 'production') {
  gatsbyConfig.plugins.push({
    resolve: `gatsby-plugin-algolia`,
    options: {
      appId: process.env.GATSBY_ALGOLIA_APP_ID,
      apiKey: process.env.GATSBY_ALGOLIA_ADMIN_KEY,
      indexName: process.env.GATSBY_ALGOLIA_INDEX_NAME,
      queries,
      chunkSize: 10000,
    },
  });
}

module.exports = gatsbyConfig;
```

Comment (richbachman) [reply packages/paste-website/gatsby-config.js:13]: Looks like I was just over-complicating things. Your solution is simpler and works.

**Pull request merged**